src/api/java/com/minecolonies/api/colony/buildings/modules/IAltersRequiredItems.java

```diff
@@ -0,0 +1,19 @@
+package com.minecolonies.api.colony.buildings.modules;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Tuple;
+
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Module type to register specific blocks to a building (beds, workstations, etc).
+ */
+public interface IAltersRequiredItems extends IBuildingModule
+{
+    /**
+     * Check if additional items have to be kept and add to map if necessary.
+     * @param toKeep the map of items that should be kept already.
```

Comment:
You might want to make this a consumer based API, where on invocation a callback is invoked with the map as a parameter.

That way it is easier to remove and add things, or even replace them.
Adds a that more seperation of concerns.